src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -63,6 +69,118 @@ private static String getFileName(String uri) {
             }
         }
 
+        private String getConfigurationUrl() {
+            YarnConfiguration conf = new YarnConfiguration();
+            String httpPolicy = conf.get(YARN_HTTP_POLICY);
+            if (httpPolicy != null && httpPolicy.equals(YARN_HTTP_POLICY_HTTPS_ONLY)) {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+                }
+                return "https://" + address + "/conf";
+            } else {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+                }
+                return "http://" + address + "/conf";
+            }
+        }
+
+        private Protos.CommandInfo getCommandInfo() {
+            MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
+            CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
+            if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
```

Comment:
Add a comment explaining that if NodeManagerUri is present, then we will use it to fetch the Hadoop binary for remote distribution mode.